### PR TITLE
Fix avatar texture loading

### PR DESCRIPTION
**Pull request by SamGondelman**
Like overlays and entities, avatars need to monitor for their textures loading in order to update their render items.

https://highfidelity.manuscript.com/f/cases/11054/Art3mis-Avatar-s-clothes-clip-into-body

Test plan:
- Start with the default wooden mannequin avatar.  Restart interface.
- Set your avatar to Yasmin.  You should not see splotches of her skin through her suit.
- Set your avatar to Art3mis.  You should not see splotches of her legs through her pants.
- Reload content several times.  Your avatar should continue to render correctly each time.

### CODE REVIEW

File: libraries/render-utils/src/MeshPartPayload.cpp [hunk 1]
```diff
@@ -480,23 +480,14 @@ ShapeKey ModelMeshPartPayload::getShapeKey() const {
...
-
-        if (_blendedVertexBuffer) {
-            batch.setInputBuffer(0, _blendedVertexBuffer, 0, sizeof(glm::vec3));
-            batch.setInputBuffer(1, _blendedVertexBuffer, _drawMesh->getNumVertices() * sizeof(glm::vec3), sizeof(glm::vec3));
-            batch.setInputStream(2, _drawMesh->getVertexStream().makeRangedStream(2));
-        } else {
-            batch.setIndexBuffer(gpu::UINT32, (_drawMesh->getIndexBuffer()._buffer), 0);
-            batch.setInputFormat((_drawMesh->getVertexFormat()));
-            batch.setInputStream(0, _drawMesh->getVertexStream());
-        }
```
Comment (SamGondelman) [inline libraries/render-utils/src/MeshPartPayload.cpp]: This is just simplifying the logic.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6081/app_12106_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6081/HighFidelity-Beta-PR12106-97c0b33975698800d3c5c66a2aa567c6931f9e51.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6081/HighFidelity-Beta-PR12106-97c0b33975698800d3c5c66a2aa567c6931f9e51.dmg

**Comment by QAadahlin** [comment]
Currently testing

**Comment by QAadahlin** [comment]
@conklin94122 Testing complete. FAILED. Skin is still visible through clothing after reloading, moving, jumping, etc. Refer to the following image.
yasmin_12106

**Comment by SamGondelman** [comment]
Just to clarify, this is what "wrong" looks like:
yasmin
If the image you posted is what you're seeing, then the fix worked.

**Comment by conklin94122** [comment]
Changed to QA approved as that was the only issue Alethia found, @divyaelango88 ok's PR.

**Pull request merged**